src/eosjs-key-conversions.ts

```diff
@@ -18,11 +18,12 @@ export const constructElliptic = (type: KeyType): EC => {
 
 export const generateKeyPair = (type: KeyType, options?: EC.GenKeyPairOptions):
     {publicKey: PublicKey, privateKey: PrivateKey} => {
-    if (process.env.EOSJS_KEYGEN_ALLOWED !== 'true') {
+    // @ts-ignore
+    if (process.env.EOSJS_KEYGEN_ALLOWED !== 'true' && window.EOSJS_KEYGEN_ALLOWED !== 'true') {
```

Comment:
All this accomplishes is making the check throw a ReferenceError in both node and browsers, you need to check existence of process and window before referencing them.

Why is this check even needed btw? elliptic.js uses cryptographically secure random generation via https://github.com/indutny/brorand

Reply:
It's not necessarily because of the cryptographic generation (which is secure), but more that client-side javascript and browsers have much less security over servers or applications on the OS.  It's the same reason why the JSSignatureProvider is not recommended over actual wallets.